tiny_cnn/core/tiny_backend.h

```diff
@@ -35,13 +35,24 @@ class tiny_backend : public math_backend {
  public:
   // context holds solution-dependent parameters
   // context should be able to hold any types of structures (like boost::any)
+  // convolution
   tiny_backend(conv_params* params,
                std::function<void(const vec_t&, int)> f1,
                std::function<void(const vec_t&, vec_t&)> f2,
                std::function<void(const vec_t&, const vec_t&, vec_t&)> f3,
-               std::vector<conv_layer_worker_specific_storage>* ptr)
-      : params_(params), conv_layer_worker_storage_(ptr),
+               std::vector<conv_layer_worker_specific_storage>* ptr1)
+      : params_(params), conv_layer_worker_storage_(ptr1),
         copy_and_pad_input(f1), copy_and_unpad_delta(f2),
+        backward_activation(f3) {}  
+
+  // deconvolution      
+  tiny_backend(conv_params* params,
+               std::function<void(const vec_t&, int)> f1,
+               std::function<void(const vec_t&, vec_t&)> f2,
+               std::function<void(const vec_t&, const vec_t&, vec_t&)> f3,
+               std::vector<deconv_layer_worker_specific_storage>* ptr1)
+      : params_(params), deconv_layer_worker_storage_(ptr1),
+        copy_and_unpad_output(f1), copy_and_pad_delta(f2),
```

Comment:
@edgarriba And here, the most important part. Is it proper to add another backend for deconvolution? I want to get access to different worker storage between convolution and deconvolution because the deconvolutional Class should have its own initiating function.